libs/langchain/langchain/vectorstores/elasticsearch.py

```diff
@@ -0,0 +1,1054 @@
+        """Create the Elasticsearch index if it doesn't already exist.
+
+        Args:
+            index_name: Name of the Elasticsearch index to create.
+            dims_length: Length of the embedding vectors.
+        """
+
+        if self.client.indices.exists(index=index_name):
+            logger.debug(f"Index {index_name} already exists. Skipping creation.")
+
+        else:
+            if dims_length is None and self.strategy.require_inference():
+                raise ValueError(
+                    "Cannot create index without specifying dims_length when the index doesn't already exist. We infer dims_length from the first embedding. Check that you have provided an embedding function."
+                )
+
+            self.strategy.before_index_setup(
+                client=self.client,
+                text_field=self.query_field,
+                vector_query_field=self.vector_query_field,
+            )
+
+            indexSettings = self.strategy.index(
+                vector_query_field=self.vector_query_field,
+                dims_length=dims_length,
+                similarity=self.distance_strategy,
+            )
+            logger.debug(
+                f"Creating index {index_name} with mappings {indexSettings['mappings']}"
+            )
+            self.client.indices.create(index=index_name, **indexSettings)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[Dict[Any, Any]]] = None,
+        ids: Optional[List[str]] = None,
+        refresh_indices: bool = True,
+        create_index_if_not_exists: bool = True,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Run more texts through the embeddings and add to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+            ids: Optional list of ids to associate with the texts.
+            refresh_indices: Whether to refresh the Elasticsearch indices
+                            after adding the texts.
+            create_index_if_not_exists: Whether to create the Elasticsearch
+                                        index if it doesn't already exist.
+
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+        """
+        try:
+            from elasticsearch.helpers import bulk
+        except ImportError:
+            raise ImportError(
+                "Could not import elasticsearch python package. "
+                "Please install it with `pip install elasticsearch`."
+            )
+
+        embeddings = []
+        ids = ids or [str(uuid.uuid4()) for _ in texts]
+        requests = []
+
+        if self.embedding is not None:
+            # If no search_type requires inference, we use the provided
+            # embedding function to embed the texts.
+            embeddings = self.embedding.embed_documents(list(texts))
```

Comment:
It's weird that texts is an iterable and the rest (ids and metadatas) are passed as lists. Not related to this PR though but I guess that add_texts is never called with large batch.